website/auth.py

```diff
@@ -35,26 +35,52 @@ def hash(password, salt):
 
-@querylog.timed
-def current_user(request):
-    if request.cookies.get(cookie_name):
-        token = DATABASE.get_token(request.cookies.get(cookie_name))
-        if token:
-            user = DATABASE.user_by_username(token['username'])
-            if user:
-                return user
-    return {'username': '', 'email': ''}
+# The current user is a slice of the user information from the database and placed on the Flask session.
+# The main purpose of the current user is to provide a convenient container for
+# * username
+# * email
+# * is_teacher
+#
+# Since the is_teacher can be changed during a session we also store a time-to-live. When retrieving the current user, we can check if we need to reload data from the database.
+#
+# The current user should be retrieved with `current_user` function since it will return a sane default.
+# You can remove the current user from the Flask session with the `forget_current_user`.
+def create_current_user(db_user):
```

Comment:
I would expect a function called `create_` to return something, but it has a side effect. Let's use a different verb?

```suggestion
def remember_current_user(db_user):
```

Reply:
It puts a tear in my eye :man_cook: :kiss: